controllers/awscluster_controller.go

```diff
@@ -155,16 +162,21 @@ func reconcileNormal(clusterScope *scope.ClusterScope) (reconcile.Result, error)
 	elbService := elb.NewService(clusterScope)
 
 	if err := ec2Service.ReconcileNetwork(); err != nil {
+		conditions.MarkFalse(awsCluster, infrav1.NetworkInfrastructureReadyCondition, infrav1.NetworkInfrastructureFailedReason, clusterv1.ConditionSeverityError, err.Error())
```

Comment:
I worry that this condition might be overly broad, I'm wondering if it would make sense to be more granular here and possibly aggregate those more granular conditions using [`conditions.SetAggregate()`](https://github.com/kubernetes-sigs/cluster-api/blob/b5962688a4b44c008cd04c138a14a76ae1f2d40c/util/conditions/setter.go#L128-L134) with the [`WithConditions()`](https://github.com/kubernetes-sigs/cluster-api/blob/b5962688a4b44c008cd04c138a14a76ae1f2d40c/util/conditions/merge_strategies.go#L39-L51) MergeOption

Reply:
The SetAggregate looks for `Ready` conditions in a set of objects and updates a condition based on their values so can't use that to aggregate. I do hear the argument for more granular reasons for the infrastructure condition, but given we also populate the message with the error on failure, what's the added value in having unique reasons?